src/faction_camp.cpp

```diff
@@ -937,7 +937,7 @@ bool talk_function::handle_camp_mission( mission_entry &cur_key, npc &p )
                     if( npc_list.empty() ) {
                         npc *comp = individual_mission( p, _( "begins to upgrade the expansion..." ),
                                                         "_faction_upgrade_exp_" + cur_key.dir, false, {},
-                                                        making->skill_used.obj().name(), making->difficulty );
+                                                        std::string( making->skill_used.obj().ident() ), making->difficulty );
```

Comment:
This can be done simpler: `making->skill_used` is already of type `skill_id`, so use it directly. And `string_id` has a function `str` which returns the id as `std::string`
```suggestion
                                                        making->skill_used.str(), making->difficulty );
```
Even better would be to change the called function to accept a `skill_id` instead of a raw string.

Reply:
Yeah, the whole system need some serious rework, but I am afraid that my coding skill is not strong enough to make that happen.